smart_open/smart_open_lib.py

```diff
@@ -984,15 +988,15 @@ def s3_iter_bucket(bucket, prefix='', accept_key=lambda key: True, key_limit=Non
 
     """
     total_size, key_no = 0, -1
-    keys = (key for key in bucket.list(prefix=prefix) if accept_key(key.name))
+    keys = ({'key': key, 'retries': retries} for key in bucket.list(prefix=prefix) if accept_key(key.name))
 
     if MULTIPROCESSING:
         logger.info("iterating over keys from %s with %i workers" % (bucket, workers))
         pool = multiprocessing.pool.Pool(processes=workers)
-        iterator = pool.imap_unordered(s3_iter_bucket_process_key, keys)
+        iterator = pool.imap_unordered(s3_iter_bucket_process_key_with_kwargs, keys)
```

Comment:
@tmylk 
Surely, but as mentioned earlier, this would cause errors in Python 2.6. Specifically, it's because partial conflicts with multiprocessing in Python 2.6: https://bugs.python.org/issue5228.

In order to make it work on Python 2.6, I've made some changes according to this thread:
https://mail.python.org/pipermail/python-dev/2009-February/086039.html